google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java

```diff
@@ -1353,4 +1355,170 @@ public void ensureFromProtoWorksWithAProxy() throws InvalidProtocolBufferExcepti
     assertFalse(comparisonFilter.isInequalityFilter());
     assertEquals(Value.newBuilder().setBooleanValue(true).build(), comparisonFilter.value);
   }
+
+  @Test
+  public void inequalityFiltersImplicitlyOrderedLexicographicallyOnCharacters() {
+    Query query_ =
+        query
+            .whereLessThan("a", "value")
+            .whereGreaterThanOrEqualTo("a", "value")
+            .whereGreaterThan("aa", "value")
+            .whereGreaterThan("b", "value")
+            .whereGreaterThan("A", "value");
+
+    List<FieldOrder> orderFields = new ArrayList<>();
+    orderFields.add(new FieldOrder("A", Query.Direction.ASCENDING));
+    orderFields.add(new FieldOrder("a", Query.Direction.ASCENDING));
+    orderFields.add(new FieldOrder("aa", Query.Direction.ASCENDING));
+    orderFields.add(new FieldOrder("b", Query.Direction.ASCENDING));
+    orderFields.add(new FieldOrder(FieldPath.documentId().toProto(), Query.Direction.ASCENDING));
```

Comment:
Just FYI, there's a function `Arrays.asList(...)` that you could use instead of a series of `orderFields.add()`s. The formatting as it is now is pretty nice and readable. So, no need to change it.